apps/snfoundry/contracts/src/Marketplace.cairo

```diff
@@ -0,0 +1,429 @@
+// Compatible with OpenZeppelin Contracts for Cairo ^0.15.0
+
+use starknet::ContractAddress;
+
+#[starknet::interface]
+pub trait IMarketplace<ContractState> {
+    fn assign_seller_role(ref self: ContractState, assignee: ContractAddress);
+    fn assign_consumer_role(ref self: ContractState, assignee: ContractAddress);
+    fn assign_admin_role(ref self: ContractState, assignee: ContractAddress);
+    fn buy_product(ref self: ContractState, token_id: u256, token_amount: u256);
+    fn buy_products(ref self: ContractState, token_ids: Span<u256>, token_amount: Span<u256>);
+    fn create_product(ref self: ContractState, initial_stock: u256, price: u256, data: Span<felt252>);
+    fn create_products(ref self: ContractState, initial_stock: Span<u256>, price: Span<u256>);
+    fn delete_product(ref self: ContractState, token_id: u256);
+    fn delete_products(ref self: ContractState, token_ids: Span<u256>);
+    fn claim(ref self: ContractState);
+}
+
+#[starknet::contract]
+mod Marketplace {
+    use starknet::event::EventEmitter;
+use openzeppelin::access::accesscontrol::AccessControlComponent;
+    use openzeppelin::access::accesscontrol::DEFAULT_ADMIN_ROLE;
+    use openzeppelin::token::erc20::interface::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use openzeppelin::token::erc1155::erc1155_receiver::ERC1155ReceiverComponent;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use openzeppelin::upgrades::UpgradeableComponent;
+    use contracts::cofi_collection::{ICofiCollectionDispatcher, ICofiCollectionDispatcherTrait};
+    use starknet::{ContractAddress, get_caller_address, get_contract_address, contract_address_const};
+    use starknet::storage::Map;
+
+    component!(path: ERC1155ReceiverComponent, storage: erc1155_receiver, event: ERC1155ReceiverEvent);
+    component!(path: SRC5Component, storage: src5, event: SRC5Event);
+    component!(path: AccessControlComponent, storage: accesscontrol, event: AccessControlEvent);
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+
+    // Role definition
+    const PRODUCER: felt252 = selector!("PRODUCER");
+    const CONSUMER: felt252 = selector!("CONSUMER");
+
+    // ERC1155Receiver
+    #[abi(embed_v0)]
+    impl ERC1155ReceiverImpl = ERC1155ReceiverComponent::ERC1155ReceiverImpl<ContractState>;
+    impl ERC1155ReceiverInternalImpl = ERC1155ReceiverComponent::InternalImpl<ContractState>;
+
+    // SRC5
+    #[abi(embed_v0)]
+    impl SRC5Impl = SRC5Component::SRC5Impl<ContractState>;
+
+    // Access Control
+    #[abi(embed_v0)]
+    impl AccessControlImpl = AccessControlComponent::AccessControlImpl<ContractState>;
+    impl AccessControlInternalImpl = AccessControlComponent::InternalImpl<ContractState>;
+    
+    // Upgradeable
+    impl UpgradeableInternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
+    const STRK_TOKEN_ADDRESS: felt252 = 0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d;
```

Comment:
Is this TOKEN_ADDRESS gonna be the same between all environments, including dev, test and production?

Reply:
yes, its the same address for mainnet and sepolia